include/picongpu/fields/MaxwellSolver/ArbitraryOrderFDTD/Weights.hpp

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2013-2020 Klaus Steiniger, Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <cmath>
+#include "picongpu/plugins/radiation/utilities.hpp"
+
+
+namespace picongpu
+{
+namespace fields
+{
+namespace maxwellSolver
+{
+namespace aoFDTD
+{
+/** Compute weights of finite differences in
+ *
+ * @tparam T_neighbors Number of neighbors used to calculate
+ *                     the derivative from finite differences.
+ *                     Order of derivative approximation is
+ *                     2 * T_neighbors
+ */
+template< uint32_t T_neighbors >
+struct AOFDTDWeights
+{
+    HDINLINE constexpr AOFDTDWeights()
+    {
+        namespace powSpace = ::picongpu::plugins::radiation::util;
+        // Set initial value
+        weights[0] = 4. * T_neighbors * 
+        powSpace::pow( (
+            factorial(2*T_neighbors) 
+            / float_X( powSpace::pow(2, 2*T_neighbors) * powSpace::pow( factorial(T_neighbors), 2) )
+        ), 2);
+
+        // Compute all other values
+        for ( uint32_t l = 1; l < T_neighbors; ++l )
+        {
+            weights[l] = -1.
+                * powSpace::pow( l - 0.5, 2 )
```

Comment:
please use `float_X` and cast `l` explicit. The second parameter of `pow` can stay integer